### PR TITLE
test: more assertions on code flow

### PR DESCRIPTION
**Pull request by danielgent**
Nuking some of my comments and making the tests more comprehensive

### CODE REVIEW

File: integration-test/flows/code-flow.spec.ts [hunk 1]
```diff
@@ -371,8 +464,108 @@ describe("code-flow", () => {
...
+        otp,
+        realm: "email",
+        username: "foo@example.com",
+      }),
+    });
+
+    expect(authRes2.status).toBe(200);
+  });
+
+  it("should not accept an invalid code", async () => {
```
Comment (danielgent) [inline integration-test/flows/code-flow.spec.ts:526]: we aren't actually testing this in unit tests

File: integration-test/flows/code-flow.spec.ts [hunk 2]
```diff
@@ -371,8 +464,108 @@ describe("code-flow", () => {
...
+      body: JSON.stringify({
+        client_id: "clientId",
+        credential_type: "http://auth0.com/oauth/grant-type/passwordless/otp",
+        otp,
+        realm: "email",
+        username: "foo@example.com",
+      }),
+    });
+
+    expect(authRes2.status).toBe(200);
```
Comment (danielgent) [inline integration-test/flows/code-flow.spec.ts:523]: maybe we don't want this going forward, but as we have this... I'll assert it :+1:

File: integration-test/flows/code-flow.spec.ts [hunk 3]
```diff
@@ -151,6 +162,88 @@ describe("code-flow", () => {
       nonce: "ehiIoMV7yJCNbSEpRq513IQgSX7XvvBM",
       iss: "https://example.com/",
     });
+
+    // ----------------------------
+    // Now log in (previous flow was signup)
```
Comment (danielgent) [inline integration-test/flows/code-flow.spec.ts:167]: Our bugs have been around sign-up vs sign-in.  The two are deceiving similar when testing *but* use very different code

**Review by markusahlstrand — APPROVED** [review]
LGTM

**Pull request merged**